src/legacy/core_plugins/kibana/public/discover/embeddable/search_embeddable.js

```diff
@@ -58,6 +60,32 @@ export class SearchEmbeddable extends Embeddable {
     };
   }
 
+  /**
+   * Generates an access URL that will be used in creating a PNG report.
+  */
+  generateAccessLink(containerState) {
+
+    const id = this.savedSearch.id;
+
+    const appState = rison.encode({ filters: containerState.filters,
+      uiState: containerState.embeddableCustomization,
+      query: containerState.query,
+    });
+
+    const globalState = rison.encode({ time: containerState.timeRange });
+
+    // Use a url service to ensure everything is escaped properly
+    const myurl = `/app/kibana#` +  url.format({
+      pathname: '/visualize/edit/' + id,
```

Comment:
This shouldn't work, the url for saved searches I think is `discover/edit` . 

Tests should probably be added as part of this PR too, so if the url format changes, the ci will fail.